odoo/modules/registry.py

```diff
@@ -118,7 +133,10 @@ def init(self, db_name):
         self._fields_by_model = None
         self._ordinary_tables = None
         self._constraint_queue = deque()
-        self.__cache = LRU(8192)
+        self.__caches = {cache_name: LRU(cache_size) for cache_name, cache_size in _REGISTRY_CACHES.items()}
+        self.__caches_groups = {}
+        for cache_name, cache in self.__caches.items():
+            self.__caches_groups.setdefault(cache_name.split('.')[0], []).append(cache)
```

Comment:
```suggestion
        self.__caches_groups = defaultdict(list)
        for cache_name, cache in self.__caches.items():
            self.__caches_groups[cache_name.split('.')[0]].append(cache)
```
And what about renaming `__cache_groups` as `__caches_by_prefix` ?